colossalai/shardformer/modeling/chatglm.py

```diff
@@ -0,0 +1,111 @@
+import torch
+import torch.nn.functional as F
+
+
+def get_flash_core_attention_forward():
+
+    from colossalai.kernel.cuda_native.flash_attention import AttnMaskType, ColoAttention
+    from tests.kit.model_zoo.transformers.chatglm2_6b.modeling_chatglm import CoreAttention
```

Comment:
You should not import any component of `tests` in colossalai.

Reply:
> You should not import any component of `tests` in colossalai.
![Uploading image.png…]()

The chatglm model is not defined in the transformers, but is defined in the pretrain weights package of "THUDM/chatglm2-6b", Therefore it cannot be imported directly. we have placed the chatglm model file under the "test.kit". The modules are all imported from "tests.kit."

Reply:
`tests` is not a valid package. If you need, you should define it in colossalai and import it in `tests.kit`

Reply:
You should never import anything defined in `tests` unless you're writing unit tests.

Reply:
> You should never import anything defined in `tests` unless you're writing unit tests.

Updated.